### ISSUE TITLE
Generic `CastInto` is not being used to create concrete implementations

### ISSUE THREAD

**Issue by tcharding** [dpc/convi#1]
I wrote this code
```rust
use convi::{CastFrom, CastInto, ExpectFrom};

fn main() {
    let a = 1_u16;

    let b = usize::cast_from(a);
    let _: usize = a.cast_into();

    let _ = u32::expect_from(b);
}
```
But got a build error
```
error[E0599]: the method `cast_into` exists for type `u16`, but its trait bounds were not satisfied
 --> src/main.rs:7:22
  |
7 |     let _: usize = a.cast_into();
  |                      ^^^^^^^^^ method cannot be called on `u16` due to unsatisfied trait bounds
  |
  = note: the following trait bounds were not satisfied:
          `u16: CastFrom<u16>`
          which is required by `u16: CastInto<u16>`
          `&u16: CastFrom<&u16>`
          which is required by `&u16: CastInto<&u16>`
          `&mut u16: CastFrom<&mut u16>`
          which is required by `&mut u16: CastInto<&mut u16>`
```

No idea why, but the generic impl is not being expanded.
```rust
impl<I> CastInto<I> for I  where I: CastFrom<Self> {
    fn cast_into(self) -> I {
        I::cast_from(self)
    }
}
```

We can see this with `cargo expand` or by the fact that I can add the following to fix the build error above and _not_ get a conflicting implementation build error.
```rust
impl CastInto<usize> for u16 {
    fn cast_into(self) -> usize  {
        usize::cast_from(self)
    }
}
```

I couldn't work out why so I'm raising an issue.

**Comment by dpc** [comment]
That's what happens when I'm in rush. Fixed in v0.0.6. :)

**Comment by tcharding** [comment]
```diff
-impl<I> CastInto<I> for I  where I: CastFrom<Self> {
+impl<F, I> CastInto<I> for F where I: CastFrom<Self> {
```

Bollocks, I stared at that thing for ages and didn't come up with this.
